packages/components/src/button/types.ts

```diff
@@ -115,23 +115,27 @@ type BaseButtonProps = {
 	 */
 	variant?: 'primary' | 'secondary' | 'tertiary' | 'link';
 	/**
-	 * Whether this is focusable.
+	 * Whether to keep the button focusable when disabled.
+	 *
+	 * @default false
 	 */
 	__experimentalIsFocusable?: boolean;
 };
 
 type _ButtonProps = {
 	/**
 	 * Whether the button is disabled.
-	 * If `true`, this will force a `button` element to be rendered.
+	 *
+	 * If `true`, this will force a `button` element to be rendered, even when an `href` is given.
 	 */
 	disabled?: boolean;
 };
 
 type AnchorProps = {
 	/**
 	 * Whether the button is disabled.
-	 * If `true`, this will force a `button` element to be rendered.
+	 *
+	 * If `true`, this will force a `button` element to be rendered, even when an `href` is given.
```

Comment:
Can this even be `true`, given that the type is `false`? Feels like this entire line is redundant here.

Reply:
Good question. In theory, yes it seems a bit wrong. In reality, the props types/descriptions for the `<a>` case and the `<button>` case are union-ed into a polymorphic type, and in most scenarios where this description can be shown to a developer I think this extra clarification is always relevant. For example, in the Storybook props table, or in IntelliSense (including parts of a codebase that are not type checked).

Basically even in an instance like `<Button href="foo" disabled={ false }>`, it's still useful to know that flipping the `disabled` to true would force the Button into a `<button>` element.